ANMN/temperature_grid_product/anmn_temp_gridded_product.py

```diff
@@ -125,195 +61,174 @@ def get_var_var_qc_in_deployment(varname, nc_file_list):
+        min_temp.append(np.min(temp[ii]))
+        
+        max_depth.append(np.max(depth[ii]))
+        min_depth.append(np.min(depth[ii]))
+        
+        max_time.append(np.max(time[ii]))
+        min_time.append(np.min(time[ii]))
+
+    max_temp = max(max_temp)
+    min_temp = min(min_temp)
+
+    max_depth = max(max_depth)
+    min_depth = min(min_depth)
+    
+    max_time = max(max_time)
+    min_time = min(min_time)
+    
+    return min_temp, max_temp, min_depth, max_depth, min_time, max_time
+
+def daterange(date1, date2, step_in_seconds):
+    for n in range(int(round(((date2 - date1).total_seconds()/step_in_seconds))) + 1):
+        yield date1 + timedelta(seconds=n*step_in_seconds)
 
 def create_time_1d(time_start, time_end, delta_in_minutes):
     """
     create a 1D time array between start and end date and a data step of delta_in_minute
+    we want this time array to be rounded to a resolution of delta_in_minute and 
+    to possibly fall on the hour 00:00:00
     """
     time_interp_array = []
-    for result in _perdelta(time_start + timedelta(minutes=delta_in_minutes)/2,
-                            time_end - timedelta(minutes=delta_in_minutes)/2,
-                            timedelta(minutes=delta_in_minutes)):
-        time_interp_array.append(result)
-
+    
+    time_start_msus = timedelta(minutes=time_start.minute, seconds=time_start.second, microseconds=time_start.microsecond)
+    time_end_msus   = timedelta(minutes=time_end.minute,   seconds=time_end.second,   microseconds=time_end.microsecond)
+    
+    time_start_rounded = datetime(time_start.year, time_start.month, time_start.day, time_start.hour)
+    time_end_rounded   = datetime(time_end.year,   time_end.month,   time_end.day,   time_end.hour)
+    
+    time_start_rounded = time_start_rounded + timedelta(seconds=np.round(time_start_msus.total_seconds()/(delta_in_minutes*60))*delta_in_minutes*60)
+    time_end_rounded   = time_end_rounded   + timedelta(seconds=np.round(time_end_msus.total_seconds()  /(delta_in_minutes*60))*delta_in_minutes*60)
+    
+    for dt in daterange(time_start_rounded, time_end_rounded, delta_in_minutes*60):
+        time_interp_array.append(dt)
+    
     return time_interp_array
 
 def create_monotonic_grid_array(nc_file_list):
     """
     create the interpolated depth and time array. The depth interpolation is 1 meter
     """
-    min_depth, max_depth = get_min_max_var_deployment(nc_file_list, 'DEPTH')
-    depth_1d_1meter      = range(min_depth, max_depth, 1)
-
-    time_start, time_end    = get_min_max_var_deployment(nc_file_list, 'TIME')
-    avrg_window, n_val_step = get_frequency_step_in_deployment(nc_file_list)
-    time_1d_interp          = create_time_1d(time_start, time_end, delta_in_minutes=avrg_window)
+    min_temp, max_temp, min_depth, max_depth, time_start, time_end = get_min_max_var_deployment(nc_file_list)
+    depth_1d_1meter = range(int(np.ceil(min_depth)), int(np.floor(max_depth)) + 1, 1)
```

Comment:
> are you wanting to exclude the min & max depths from the range?

Yes, if they're not integer values (default vertical resolution is 1m).
As an example, if min_depth = 10.1 and max_depth = 22.7, I don't want my grid to start with 10 all the way to 23 because I know that with linear interpolation and NaN outsidemin/max then I will always get NaNs at 10 and 23. So the useful range is going to be [11; 22], within that range interpolation should give me some non NaN values.